gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -103,6 +103,10 @@ public synchronized void setActive(boolean isActive) {
       return;
     }
 
+    // Need to set active first; otherwise in the STANDBY->ACTIVE transition,
+    // the onAddSpec will forward specs to the leader, which is itself.
+    this.isActive = isActive;
```

Comment:
What happens in the ACTIVE -> STANDBY transition. If this.isActive is set to false here then won't the onDeleteSpec() below get forwarded instead of resulting in cleanup in the current process?